### ISSUE TITLE
Unnecessary body update after text_content

### ISSUE THREAD

**Issue by sevenmaxis** [hopsoft/stimulus_reflex#186]
I was following the tutorial [Twitter clone](https://www.youtube.com/watch?v=F5hA79vKE_E)

And I've noticed it updates the entire page after `text_content` operation
The first part of the log is where text_content happening and the second part is where the body update occurs.

```
StimulusReflex::Channel#receive({"target"=>"PostsReflex#like", "args"=>[], "url"=>"http://localhost:3000/posts", "attrs"=>{"class"=>"card-link", "data-reflex"=>"click->PostsReflex#like", "data-id"=>"25", "href"=>"#", "data-controller"=>"stimulus-reflex", "data-action"=>"click->stimulus-reflex#__perform", "checked"=>false, "selected"=>false, "tag_name"=>"A"}, "selectors"=>[], "permanent_attribute_name"=>"data-reflex-permanent", "reflexId"=>"4a031db8-a943-4e74-888a-518ffc4f9271"})
  Post Load (0.3ms)  SELECT "posts".* FROM "posts" WHERE "posts"."id" = ? LIMIT ?  [["id", 25], ["LIMIT", 1]]
  ↳ app/reflexes/posts_reflex.rb:7:in `like'
  Post Update All (1.9ms)  UPDATE "posts" SET "likes_counter" = COALESCE("likes_counter", 0) + ? WHERE "posts"."id" = ?  [["likes_counter", 1], ["id", 25]]
  ↳ app/reflexes/posts_reflex.rb:8:in `like'
[ActionCable] Broadcasting to timeline: {"cableReady"=>true, "operations"=>{"textContent"=>[{"selector"=>"post-25-likes", "text"=>8}]}}
Processing by PostsController#index as HTML
TimeLineChannel transmitting {"cableReady"=>true, "operations"=>{"textContent"=>[{"selector"=>"post-25-likes", "text"=>8}]}} (via streamed from timeline)
TimeLineChannel transmitting {"cableReady"=>true, "operations"=>{"textContent"=>[{"selector"=>"post-25-likes", "text"=>8}]}} (via streamed from timeline)
  Rendering posts/index.html.erb within layouts/application
  Rendered posts/_form.html.erb (Duration: 1.0ms | Allocations: 399)
  Post Load (0.2ms)  SELECT "posts".* FROM "posts" ORDER BY "posts"."created_at" DESC
  ↳ app/views/posts/index.html.erb:8
  Rendered collection of posts/_post.html.erb [25 times] (Duration: 8.0ms | Allocations: 5412)
  Rendered posts/index.html.erb within layouts/application (Duration: 10.7ms | Allocations: 6849)
[Webpacker] Everything's up-to-date. Nothing to do
Completed 200 OK in 19ms (Views: 17.8ms | ActiveRecord: 0.2ms | Allocations: 10409)


[ActionCable] Broadcasting to StimulusReflex::Channel:4d54b9ceb075ac58a966089beaa39a17: {"cableReady"=>true, "operations"=>{"morph"=>[{"selector"=>"body", "html"=>"\n    <div class=\"container\">\n  <h1>Chatter</h1>\n  <h4>CableReady &amp; StimulusReflex demo</h4>\n\n  <form class=\"my-4\" action=\"/posts\" accept-charset=\"UTF-8\" data-remote=\"true\" method=\"post\">\n<input type=\"hidden\" name=\"authenticity_token\" value=\"rEhQ5p8BlwDHXwqdbOSss4sI/wnFuPhesPhLMtliewCUE8fZvNiMNhBT5syQSEHQ+RX/WGFykYI5NSGcVc/wfw==\">\n  <div class=\"form-group\">\n    <textarea placeholder=\"Enter your message\" class=\"form-control\" rows=\"3\" name=\"post[body]\" id=\"post_body\">\n</textarea>\n  </div>\n  <div class=\"actions\">\n    <input type=\"submit\" name=\"commit\" value=\"Create Post\" class=\"btn btn-primary\" data-disable-with=\"Create Post\">\n  </div>\n</form>\n\n  <div id=\"timeline\">\n    <div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- 6 minutes ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">ppp</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"25\" href=\"#\">\n      Likes (<span id=\"post-25-likes\">8</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"25\" href=\"#\">\n      Reposts (<span id=\"post-25-reposts\">11</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- 6 minutes ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">lll</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"24\" href=\"#\">\n      Likes (<span id=\"post-24-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"24\" href=\"#\">\n      Reposts (<span id=\"post-24-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- 7 minutes ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">kkk</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"23\" href=\"#\">\n      Likes (<span id=\"post-23-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"23\" href=\"#\">\n      Reposts (<span id=\"post-23-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- 9 minutes ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">zzz</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"22\" href=\"#\">\n      Likes (<span id=\"post-22-likes\">9</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"22\" href=\"#\">\n      Reposts (<span id=\"post-22-reposts\">2</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">yyy</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"21\" href=\"#\">\n      Likes (<span id=\"post-21-likes\">34</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"21\" href=\"#\">\n      Reposts (<span id=\"post-21-reposts\">17</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">zzz</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"20\" href=\"#\">\n      Likes (<span id=\"post-20-likes\">1</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"20\" href=\"#\">\n      Reposts (<span id=\"post-20-reposts\">6</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">xxx</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"19\" href=\"#\">\n      Likes (<span id=\"post-19-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"19\" href=\"#\">\n      Reposts (<span id=\"post-19-reposts\">6</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sss</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"18\" href=\"#\">\n      Likes (<span id=\"post-18-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"18\" href=\"#\">\n      Reposts (<span id=\"post-18-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sss</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"17\" href=\"#\">\n      Likes (<span id=\"post-17-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"17\" href=\"#\">\n      Reposts (<span id=\"post-17-reposts\">1</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 1 hour ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sss</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"16\" href=\"#\">\n      Likes (<span id=\"post-16-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"16\" href=\"#\">\n      Reposts (<span id=\"post-16-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">hghh</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"15\" href=\"#\">\n      Likes (<span id=\"post-15-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"15\" href=\"#\">\n      Reposts (<span id=\"post-15-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">rrr</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"14\" href=\"#\">\n      Likes (<span id=\"post-14-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"14\" href=\"#\">\n      Reposts (<span id=\"post-14-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">zzz</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"13\" href=\"#\">\n      Likes (<span id=\"post-13-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"13\" href=\"#\">\n      Reposts (<span id=\"post-13-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">yyy</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"12\" href=\"#\">\n      Likes (<span id=\"post-12-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"12\" href=\"#\">\n      Reposts (<span id=\"post-12-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">xxx</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"11\" href=\"#\">\n      Likes (<span id=\"post-11-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"11\" href=\"#\">\n      Reposts (<span id=\"post-11-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">s</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"10\" href=\"#\">\n      Likes (<span id=\"post-10-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"10\" href=\"#\">\n      Reposts (<span id=\"post-10-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">33</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"9\" href=\"#\">\n      Likes (<span id=\"post-9-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"9\" href=\"#\">\n      Reposts (<span id=\"post-9-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">z</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"8\" href=\"#\">\n      Likes (<span id=\"post-8-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"8\" href=\"#\">\n      Reposts (<span id=\"post-8-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">y</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"7\" href=\"#\">\n      Likes (<span id=\"post-7-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"7\" href=\"#\">\n      Reposts (<span id=\"post-7-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">x</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"6\" href=\"#\">\n      Likes (<span id=\"post-6-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"6\" href=\"#\">\n      Reposts (<span id=\"post-6-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sdfdsf</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"5\" href=\"#\">\n      Likes (<span id=\"post-5-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"5\" href=\"#\">\n      Reposts (<span id=\"post-5-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sdfds</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"4\" href=\"#\">\n      Likes (<span id=\"post-4-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"4\" href=\"#\">\n      Reposts (<span id=\"post-4-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 3 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sfds</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"3\" href=\"#\">\n      Likes (<span id=\"post-3-likes\">0</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"3\" href=\"#\">\n      Reposts (<span id=\"post-3-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 4 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">dfds\n</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"2\" href=\"#\">\n      Likes (<span id=\"post-2-likes\">3</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"2\" href=\"#\">\n      Reposts (<span id=\"post-2-reposts\">0</span>)\n</a>  </div>\n</div>\n<div class=\"card mb-2\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title text-muted\">\n      Serhii\n      <small>- about 4 hours ago</small>\n    </h5>\n    <div class=\"card-text lead mb-2\">sfsdf</div>\n    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#like\" data-id=\"1\" href=\"#\">\n      Likes (<span id=\"post-1-likes\">7</span>)\n</a>    <a class=\"card-link\" data-reflex=\"click-&gt;PostsReflex#repost\" data-id=\"1\" href=\"#\">\n      Reposts (<span id=\"post-1-reposts\">4</span>)\n</a>  </div>\n</div>\n\n  </div>\n</div>\n\n  ", "childrenOnly"=>true, "permanentAttributeName"=>"data-reflex-permanent", "stimulusReflex"=>{"target"=>"PostsReflex#like", "args"=>[], "url"=>"http://localhost:3000/posts", "attrs"=>{"class"=>"card-link", "data-reflex"=>"click->PostsReflex#like", "data-id"=>"25", "href"=>"#", "data-controller"=>"stimulus-reflex", "data-action"=>"click->stimulus-reflex#__perform", "checked"=>false, "selected"=>false, "tagName"=>"A"}, "selectors"=>["body"], "permanentAttributeName"=>"data-reflex-permanent", "reflexId"=>"4a031db8-a943-4e74-888a-518ffc4f9271", "last"=>true}}]}}
StimulusReflex::Channel transmitting {"cableReady"=>true, "operations"=>{"morph"=>[{"selector"=>"body", "html"=>"\n    <div class=\"container\">\n  <h1>Chatter</h1>\n  <h4>CableReady &amp; StimulusReflex demo</h4>\n\n  <form class=\"my-4\" action=\"/posts\" accept-charset=\"UTF-8\" data-remote=\"true\" method=\"post\">\n<input type=... (via streamed from StimulusReflex::Channel:4d54b9ceb075ac58a966089beaa39a17)
StimulusReflex::Channel transmitting {"cableReady"=>true, "operations"=>{"morph"=>[{"selector"=>"body", "html"=>"\n    <div class=\"container\">\n  <h1>Chatter</h1>\n  <h4>CableReady &amp; StimulusReflex demo</h4>\n\n  <form class=\"my-4\" action=\"/posts\" accept-charset=\"UTF-8\" data-remote=\"true\" method=\"post\">\n<input type=... (via streamed from StimulusReflex::Channel:4d54b9ceb075ac58a966089beaa39a17)
```

**Comment by leastbad** [comment]
We’d love to implement something equivalent to the LiveView .leex templates, but it is a non-trivial undertaking.

Please do investigate!

**Comment by hopsoft** [comment]
Actually, you could handle this scenario in an interesting way after 3.2.0 ships.

1. Trigger a reflex in the standard way
1. Setup the `text_content` CableReady broadcast and broadcast from within a `before_reflex` callback
1. Halt the the callback chain to force a noop and prevent page re-render for the current user (this can be accomplished with a `throw :abort` in the `before_reflex` callback method)

**Comment by hopsoft** [comment]
Note that #211 will provide the flexibility to perform such updates without redundant morphs etc...

**Comment by leastbad** [comment]
Great news: #211 got merged and we're on the verge of launching v3.3, which will support [Morph Modes](https://docs.stimulusreflex.com/morph-modes#selector-morphs).
